profile.go

```diff
@@ -0,0 +1,415 @@
+package lipgloss
+
+import (
+	"image/color"
+	"math"
+	"strconv"
+	"strings"
+
+	"github.com/charmbracelet/x/exp/term/ansi"
+	"github.com/lucasb-eyer/go-colorful"
+)
+
+// Profile is a color profile: Ascii, ANSI, ANSI256, or TrueColor.
+type Profile int
+
+const (
+	// NoTTY, no terminal profile
+	NoTTY Profile = iota
+	// Ascii, uncolored profile
+	Ascii // nolint: revive
+	// ANSI, 4-bit color profile
+	ANSI
+	// ANSI256, 8-bit color profile
+	ANSI256
+	// TrueColor, 24-bit color profile
+	TrueColor
+)
+
+// Convert transforms a given Color to a Color supported within the Profile.
+func (p Profile) Convert(c ansi.Color) ansi.Color {
+	if p <= Ascii {
+		return noColor
+	}
+
+	switch v := c.(type) {
+	case ansi.BasicColor:
+		return v
+
+	case ansi.ExtendedColor:
+		if p == ANSI {
+			return ansi256ToANSIColor(v)
+		}
+		return v
+
+	case ansi.TrueColor, color.Color:
+		h, ok := colorful.MakeColor(v)
+		if !ok {
+			return nil
+		}
+		if p != TrueColor {
+			ac := hexToANSI256Color(h)
+			if p == ANSI {
+				return ansi256ToANSIColor(ac)
+			}
+			return ac
+		}
+		return v
+	}
+
+	return c
+}
+
+// Color creates a Color from a string. Valid inputs are hex colors, as well as
+// ANSI Color codes (0-15, 16-255).
+func (p Profile) Color(s string) ansi.Color {
+	if len(s) == 0 {
+		return noColor
+	}
+
+	var c ansi.Color
+	if strings.HasPrefix(s, "#") {
```

Comment:
I think go-colorful (which is already a dep) can do this as well? maybe remove go-colorful, or use it here too...

Reply:
Good point